### PR TITLE
[chore] prepare release 0.14.0

### PR DESCRIPTION
**Pull request by odubajDT**

### CODE REVIEW

**Review by dyladan — APPROVED** [review]

File: internal/tools/go.mod [hunk 1]
```diff
@@ -7,7 +7,6 @@ toolchain go1.23.1
 require (
 	github.com/goreleaser/goreleaser/v2 v2.3.1
 	github.com/jstemmer/go-junit-report v1.0.0
-	github.com/jstemmer/go-junit-report/v2 v2.1.0
```
Comment (dyladan) [inline internal/tools/go.mod:10]: Why was this removed?

**Review by mowies — APPROVED** [review]

File: internal/tools/go.mod (hunk 1 above)
Comment (odubajDT) [reply internal/tools/go.mod:10]: `go mod tidy` was executed just to clean up unused dependencies, this dependency is not used right now at all, but we will try to introduce it via https://github.com/Dynatrace/dynatrace-otel-collector/pull/298

It's has zero importance for the release

**Review by pirgeo — APPROVED** [review]
